src/Chainweb/Version.hs

```diff
@@ -554,15 +563,18 @@ window FastTimedCPM{} = Nothing
 window Development = Just $ WindowWidth 120
 -- 120 blocks, should take 1 hour given a 30 second BlockRate.
 window Testnet02 = Just $ WindowWidth 120
-
-txSilenceDates :: ChainwebVersion -> Maybe (Time Micros)
-txSilenceDates Test{} = Nothing
-txSilenceDates TimedConsensus{} = Nothing
-txSilenceDates PowConsensus{} = Nothing
-txSilenceDates TimedCPM{} = Nothing
-txSilenceDates FastTimedCPM{} = Nothing
-txSilenceDates Development = Nothing
-txSilenceDates Testnet02 = Nothing
+window Mainnet01 = Just $ WindowWidth 120
+
+txSilenceEndDate :: ChainwebVersion -> Maybe (Time Micros)
+txSilenceEndDate Test{} = Nothing
+txSilenceEndDate TimedConsensus{} = Nothing
+txSilenceEndDate PowConsensus{} = Nothing
+txSilenceEndDate TimedCPM{} = Nothing
+txSilenceEndDate FastTimedCPM{} = Nothing
+txSilenceEndDate Development = Nothing
+txSilenceEndDate Testnet02 = Nothing
+-- Thursday, 2019 December 5, 12:00 AM
+txSilenceEndDate Mainnet01 = Just . Time $ TimeSpan 1575504000000000
```

Comment:
We need the code for these time values to be human-readable, not comments

Reply:
This code is executed on each block validation. Parsing time values is expensive. Let's make sure that the implementation is efficient (e.g. by caching the result).

Reply:
Also, if we use something human readable, we should use a standardized format, that is unambiguous with respect to the time system. In block creation times (and this time value is compared with block creation times) we use SI seconds since POSIX epoch. 

So, it should be clear how a the human readable format is converted to that standard.

Reply:
See comment above.

Reply:
If it has to be human readable, I propose that we use ISO 8601 format `yyyy-mm-ddTHH:MM:SS+00:00`. @fosskers is using `NOINLINE` enough to ensure that it's not parsed each time it is used? Otherwise, with template Haskell it could be reduced to a numeric constant at compile time and inlined (`Time` and `TimeSpan` are both newtypes).

When comparing POSIX seconds with current times, the system clock is used which is usually based on UTC. That means, that we ignore leap seconds and do conversions assuming that each day has 86400 seconds. (I would prefer to convert based on TAI seconds, but most systems don't provide a clock for that. Also UTC is much more common.) Therefore, @slpopejoy has a point here, that providing the constant in UTC format is more explicit about the time standard we are using.